### ISSUE TITLE
Gitpod: Frontend cannot connect to trace server

### ISSUE THREAD

**Issue by ebugden** [theia-ide/theia-trace-extension#488]
Can be observed by opening the merging branch in PR #487.

The server seems to start correctly (see below), but the frontend fails to connect to it (we know this because it tries to launch a new server on 8080 which fails because the original server is still running on 8080).

![image](https://user-images.githubusercontent.com/28311615/133817610-37b2b6b2-9b04-4ca4-be42-0fd68de96d2c.png)

https://user-images.githubusercontent.com/28311615/133816015-eada6cca-85b2-4562-9dce-e7c4c78c3a06.mp4

## The server's port (8080) is public
![image](https://user-images.githubusercontent.com/28311615/133815412-b77a8cf0-0fd3-46ab-ad71-4725a28fccfc.png)

## The server seems to start correctly
The terminal output is the same as before the Theia upgrade.

### Server start log before Theia version upgrade (gitpod still works):
![image](https://user-images.githubusercontent.com/28311615/133809652-f32a0e7a-7228-448c-b5ad-8ca7ba843478.png)

### Server start log now: 
![image](https://user-images.githubusercontent.com/28311615/133813279-f89aecbe-856a-419d-b725-7cd553d7e2ce.png)

**Comment by paul-marechal** [comment]
It looks like you are making a request from the frontend (your browser) on `localhost:8080`. In this case, even if your page is opened on some gitpod workspace weird origin, I believe that `localhost` still refers to _your_  local machine.

A potential fix would be to define the URL to the trace server as something like:

```ts
const traceUrl = new URL(window.location.origin);
traceUrl.port = 8080;
// ...
fetch(traceUrl.href, ...);
```

This way the current's Theia frontend origin will be reused, only changing the port to reach the trace server.

I assumed that Gitpod serves ports on the same origin, but if that's not the case then the issue becomes identifying what's the place where Gitpod serves the trace server.

If this used to work before then I'm confused and I am most certainly missing something, but this all I can think of atm.

**Comment by paul-marechal** [comment]
Ah looks like Gitpod does things like `https://<port>-<workspace-origin>/...` to serve services, so my snippet wouldn't work.

**Comment by paul-marechal** [comment]
Ok, so there is code to detect what's the trace server Gitpod URL:

```bash
export GITPOD_URL=$(gp url 8080) && export TRACE_SERVER_URL=$(echo "$GITPOD_URL/tsp/api") && yarn start:browser ../../TraceCompassTutorialTraces/ --hostname=0.0.0.0
```

Then this should be picked up by the frontend code trying to reach the trace server:

https://github.com/theia-ide/theia-trace-extension/blob/fa979baf5d51c4e5243c0c86f8ee62ce438fbb04/theia-extensions/viewer-prototype/src/common/trace-viewer-environment.ts#L37-L38

So you get an error on Gitpod where the request is made to `localhost` it means something went sideways and the Gitpod URL is not correctly being passed around?

**Comment by IbrahimFradj** [comment]
gitpod doesn't work for me either, but when I run it directly from my computer, I get the same error for the server, but I manage to launch the application and open the views. 

![bug server](https://user-images.githubusercontent.com/83769103/133830344-4a788de4-eab1-45c2-8d2a-b02cc1b44a5f.gif)

**Comment by paul-marechal** [comment]
@IbrahimFradj either your Java environment is causing the server to fail to start, or the latest jar is bogus?

**Comment by arfio** [comment]
Yes I have this error when downloading the latest Trace Compass server with Theia-trace-extension, and it seems to be an error with the logger, in any case, the server continues to work normally even with this error.

**Comment by ebugden** [comment]
The server's complaints about failing to load things on startup are likely unrelated since the complaints were there back when the server connection still worked before the Theia update (see issue description).

Since this server connection issue is not present on local installations, it makes me think it's a permissions or export issue. I'm going to look into areas @paul-marechal:
* [gitpod.yml: line 15 ](https://github.com/theia-ide/theia-trace-extension/blob/bbd7afca8d923b1fd8064de7df45ec256e846b28/.gitpod.yml#L25)
* [trace-viewer-environment.ts: lines 36-37](https://github.com/theia-ide/theia-trace-extension/blob/fa979baf5d51c4e5243c0c86f8ee62ce438fbb04/theia-extensions/viewer-prototype/src/common/trace-viewer-environment.ts#L37-L38)

**Comment by bhufmann** [comment]
I created #493 to track the fix for connecting the GitPod trace server instead of local host. 

What I noticed is that the with the latest code, the `traceServerUrl` is set with the default value before the environment variable TRACE_SERVER_URL is actually read. This seems to happen because at startup of the extension a preference change event is created for the port and then the traceServerUrl is set. The start-up procedure / start-up order seems to be different with Theia 1.17.2 than with older versions.

I'll provide a PR for that.
